### PR TITLE
[Asmjit ] Asmjit update - Following pull request #39090

### PR DESCRIPTION
**Pull request by JakubMelka**
If this PR updates an existing port, please uncomment and fill out this checklist:

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by JakubMelka** [comment]
@microsoft-github-policy-service agree

**Comment by JakubMelka** [comment]
This request is following pull request #39090.

File: ports/asmjit/vcpkg.json [hunk 1]
```diff
@@ -1,9 +1,10 @@
 {
   "name": "asmjit",
-  "version-date": "2023-03-25",
-  "description": "Complete x86/x64 JIT and Remote Assembler for C++",
+  "version-date": "2024-06-05",
```
Comment (kobalicek) [inline ports/asmjit/vcpkg.json:3]: Here must be a correct version: 2024-06-28

File: versions/a-/asmjit.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "6e65362a730457a3168f8a7ad59358472eed0fcd",
+      "version-date": "2024-06-05",
```
Comment (kobalicek) [inline versions/a-/asmjit.json:5]: Here must be a correct version: 2024-06-28

File: versions/baseline.json [hunk 3]
```diff
@@ -269,7 +269,7 @@
       "port-version": 7
     },
     "asmjit": {
-      "baseline": "2023-03-25",
+      "baseline": "2024-06-05",
```
Comment (kobalicek) [inline versions/baseline.json:272]: Here must be a correct version: 2024-06-28

File: ports/asmjit/vcpkg.json (hunk 1 above)
Comment (JakubMelka) [reply ports/asmjit/vcpkg.json]: Fixed.

File: versions/a-/asmjit.json (hunk 2 above)
Comment (JakubMelka) [reply versions/a-/asmjit.json]: Fixed date, but @kobalicek, I think "git-tree" should also be updated? How this hash is constructed?

Comment (JakubMelka) [reply versions/a-/asmjit.json]: Figured it out. ./vcpkg x-add-version --all --overwrite-version solved it.

File: ports/asmjit/vcpkg.json [hunk 4]
```diff
@@ -1,9 +1,10 @@
 {
   "name": "asmjit",
-  "version-date": "2023-03-25",
-  "description": "Complete x86/x64 JIT and Remote Assembler for C++",
+  "version-date": "2024-06-28",
+  "description": "AsmJit is a lightweight library for machine code generation written in C++ language",
   "homepage": "https://github.com/asmjit/asmjit",
-  "supports": "!arm",
+  "documentation": "https://asmjit.com/doc/index.html",
```
Comment (JonLiu1993) [inline ports/asmjit/vcpkg.json:6]: @JakubMelka, Thanks for your PR, there are two points to note here. First, we prefer the homepage URL of github `https://github.com/asmjit/asmjit`
Second, can you explain why the new version does not support arm?

File: ports/polyhook2/portfile.cmake [hunk 5]
```diff
@@ -1,10 +1,9 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO stevemk14ebr/PolyHook_2_0
-    REF fd2a88f09c8ae89440858fc52573656141013c7f
-    SHA512 09eef1b0ca75f9420fa258c4290b6b62a3115531e77c91f9d223b2c4a10e8ee95b243bf9f6598d7c55b76ca4e4a185cdf1231b40d5ef87d7d405b8b434eb6cb0
+    REF 8cd6cb4ef0f2a599f35bdd46ac5833843aea6523
+    SHA512 93e6fcbb47f4b894c325766e1b2c146cebfe73424473ce0f52c8c923bd5b1276aebb3af03da9fc7027e2027994663c47a5a4f5a4f3c4a997a322886ac43401ec
     HEAD_REF master
-    PATCHES fix-dep.patch
```
Comment (JonLiu1993) [inline ports/polyhook2/portfile.cmake:7]: The patch has been applied to the latest version.

File: versions/f-/fbgemm.json [hunk 6]
```diff
@@ -1,7 +1,12 @@
 {
   "versions": [
     {
-      "git-tree": "e19b797fb42fad6dae0f9aca0890ddc62535800f",
```
Comment (JonLiu1993) [inline versions/f-/fbgemm.json:4]: Restore it,  please do not modify the history `git-tree`.

**Comment by JonLiu1993** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: ports/asmjit/vcpkg.json (hunk 4 above)
Comment (dg0yt) [reply ports/asmjit/vcpkg.json:6]: @JonLiu1993 You read it wrong, twice.
The homepage URL is *unchanged*.
Arm support is *added*, not removed.

File: ports/fbgemm/vcpkg.json [hunk 7]
```diff
@@ -1,10 +1,11 @@
 {
   "name": "fbgemm",
   "version": "0.4.1",
+  "port-version": 1,
   "description": "FB (Facebook) + GEMM (General Matrix-Matrix Multiplication)",
   "homepage": "https://code.fb.com/ml-applications/fbgemm/",
   "license": "BSD-3-Clause",
-  "supports": "!(x86 | uwp)",
+  "supports": "!arm & !x86 & !uwp",
```
Comment (dg0yt) [inline ports/fbgemm/vcpkg.json:8]: *This* exclusion of arm might have been hitten behind a `asmjit` failure cascade. So it is not worse than before.

File: versions/f-/fbgemm.json (hunk 6 above)
Comment (dg0yt) [reply versions/f-/fbgemm.json:4]: This is badly displayed on Github. Let me add a suggestion a feew lines later...

File: versions/f-/fbgemm.json [hunk 8]
```diff
@@ -1,7 +1,12 @@
...
+      "git-tree": "27b4707b88bb2ed527d372438c406d1a97323229",
+      "version": "0.4.1",
+      "port-version": 1
+    },
+    {
+      "git-tree": "2aa09f4d6344a0c379f6b65fae72dc8117759805",
```
Comment (dg0yt) [inline versions/f-/fbgemm.json:9]: Here:
```suggestion
      "git-tree": "e19b797fb42fad6dae0f9aca0890ddc62535800f",
```
i.e. the original value for `0.4.1#0`.

**Review by dg0yt — CHANGES_REQUESTED** [review]
Just one change needed. Thanks from the community.

File: ports/asmjit/vcpkg.json (hunk 4 above)
Comment (dg0yt) [reply ports/asmjit/vcpkg.json:6]: To be honest, my preference is  
homepage: https://asmjit.com/
documentation: https://asmjit.com/doc/

https://github.com/asmjit/asmjit *is not* the homepage.

File: ports/polyhook2/portfile.cmake (hunk 5 above)
Comment (JakubMelka) [reply ports/polyhook2/portfile.cmake:7]: Is some action from me needed?

File: versions/f-/fbgemm.json (hunk 6 above)
Comment (JakubMelka) [reply versions/f-/fbgemm.json:4]: This is very strange. This is commit from other author. In diff, it is displayed as this: 
image

You mean that I should restore hash for port-version 0?

File: ports/fbgemm/vcpkg.json (hunk 7 above)
Comment (JakubMelka) [reply ports/fbgemm/vcpkg.json:8]: Yes, ARM support for ASMJIT has been added, but FBGEMM does not support it.

File: versions/f-/fbgemm.json (hunk 8 above)
Comment (JakubMelka) [reply versions/f-/fbgemm.json]: OK, I have reverted it.

File: versions/f-/fbgemm.json (hunk 6 above)
Comment (JakubMelka) [reply versions/f-/fbgemm.json:4]: port-version 0 has been restored.

**Comment by JakubMelka** [comment]
@JonLiu1993, I hope I have fixed all your issues.

**Review by dg0yt — APPROVED** [review]
LGTM, but someone else must approve.

**Comment by kobalicek** [comment]
I approve, although my approval is worthless here :)

And I don't approve the patching of course, but that we know from the previous PR.

**Pull request merged**